### PR TITLE
update configuration to YAML

### LINKED ISSUE #797: update configuration to YAML
# Description
The current pycsw [configuration](https://docs.pycsw.org/en/latest/configuration.html) is based on the Windows ini format, which supports one level of grouping/nesting.

Updating the pycsw configuration to YAML will provide more flexibility for nesting, as well as the ability for better cardinality.

Notes:
- we may want to consider the pygeoapi (YAML) configuration for interoperability's sake
- we will likely need a migration utility via `pycsw-admin.py`
- this would be breaking change for the 3.0 release target

# Environment

- operating system: all
- Python version: all
- pycsw version: master
- source/distribution
  - [x] git clone
  - [ ] DebianGIS/UbuntuGIS
  - [ ] PyPI
  - [ ] zip/tar.gz
  - [ ] other (please specify):
- web server
  - [ ] Apache/mod_wsgi
  - [ ] CGI
  - [x] other (please specify): Flask

# Steps to Reproduce
`cat default.cfg` to view current configuration
# Additional Information
None

### PR DESCRIPTION
**Pull request by tomkralidis**
# Overview
Updates pycsw configuration from ini to YAML.

# Related Issue / Discussion
Fixes #797
# Additional Information

### TODO

- [x] migration tool/script

### External updates required
- [x] pycsw public demo (https://github.com/geopython/demo.pycsw.org)
- [ ] GeoNode CSW plugin
- [ ] CKAN CSW plugin

# Contributions and Licensing

(as per https://github.com/geopython/pycsw/blob/master/CONTRIBUTING.rst#contributions-and-licensing)

- [x] I'd like to contribute [feature X|bugfix Y|docs|something else] to pycsw. I confirm that my contributions to pycsw will be compatible with the pycsw license guidelines at the time of contribution.
- [x] I have already previously agreed to the pycsw Contributions and Licensing Guidelines

### CODE REVIEW

**Comment by kalxas** [comment]
We will also need to update the kubernetes helm charts when this is merged

**Comment by tomkralidis** [comment]
k8s and helm charts first pass updated as part of this PR.

**Comment by tomkralidis** [comment]
- migration script added
- associated demo update PR in https://github.com/geopython/demo.pycsw.org/pull/10

File: docker/helm/templates/pycsw-configmap.yaml [hunk 1]
```diff
@@ -1,71 +1,85 @@
...
-{{ if .Values.pycsw.config.inspire.enabled }}    enabled={{ .Values.pycsw.config.inspire.enabled }}
-{{ end }}{{ if .Values.pycsw.config.inspire.languages_supported }}    languages_supported={{ .Values.pycsw.config.inspire.languages_supported }}
-{{ end }}{{ if .Values.pycsw.config.inspire.default_language }}    default_language={{ .Values.pycsw.config.inspire.default_language }}
-{{ end }}{{ if .Values.pycsw.config.inspire.date }}    date={{ .Values.pycsw.config.inspire.date }}
-{{ end }}{{ if .Values.pycsw.config.inspire.gemet_keywords }}    gemet_keywords={{ .Values.pycsw.config.inspire.gemet_keywords }}
-{{ end }}{{ if .Values.pycsw.config.inspire.conformity_service }}    conformity_service={{ .Values.pycsw.config.inspire.conformity_service }}
-{{ end }}{{ if .Values.pycsw.config.inspire.contact_name }}    contact_name={{ .Values.pycsw.config.inspire.contact_name }}
-{{ end }}{{ if .Values.pycsw.config.inspire.contact_email }}    contact_email={{ .Values.pycsw.config.inspire.contact_email }}
-{{ end }}{{ if .Values.pycsw.config.inspire.temp_extent }}    temp_extent={{ .Values.pycsw.config.inspire.temp_extent }}
+    repositor:
```
Comment (kalxas) [inline docker/helm/templates/pycsw-configmap.yaml:78]: typo: should be repository

File: docker/helm/templates/pycsw-configmap.yaml [hunk 2]
```diff
@@ -1,71 +1,85 @@
...
+    manager:
+{{ if .Values.pycsw.config.manager.transactions }}    transactions: {{ .Values.pycsw.config.manager.transactions }}
+{{ end }}{{ if .Values.pycsw.config.manager.allowed_ips }}    allowed_ips: {{ .Values.pycsw.config.manager.allowed_ips }}
+{{ end }}{{ if .Values.pycsw.config.manager.csw_harvest_pagesize }}    csw_harvest_pagesize: {{ .Values.pycsw.config.manager.csw_harvest_pagesize }}
+{{ end }}
+    metadata:
+        identification:
+{{ if .Values.pycsw.config.metadata.identification.title }}        title: {{ .Values.pycsw.config.metadata.identification.title }}
+{{ end }}{{ if .Values.pycsw.config.metadata.identification.description }}        description: {{ .Values.pycsw.config.metadata.identification.description }}
+{{ end }}{{ if .Values.pycsw.config.metadata.identification.keywords }}        keywords: {{ .Values.pycsw.config.metadata.identification.keywords }}
```
Comment (kalxas) [inline docker/helm/templates/pycsw-configmap.yaml:42]: Need to test if keywords are dumped in yaml style here

**Review by kalxas — CHANGES_REQUESTED** [review]
Excellent work!

File: docker/helm/templates/pycsw-configmap.yaml (hunk 1 above)
Comment (kalxas) [reply docker/helm/templates/pycsw-configmap.yaml]: Fixed

**Review by kalxas — APPROVED** [review]

**Comment by pvgenuchten** [comment]
Nice work, just to confirm: the override mechanism with environmental variables is still supported?

**Comment by tomkralidis** [comment]
@pvgenuchten yes, environment variables are interpreted as per before.

**Pull request merged**